### DISCUSSION TITLE
Problem with typeclasses blog post

### DISCUSSION THREAD

**Discussion by bon** [coalton-lang/coalton#352] [Help!] [answered]
I was reading through the [blog post on typeclasses](https://coalton-lang.github.io/20211212-typeclasses) but ran into a problem with the `coalton` and `cl` packages. Here's a snippet that fails with the error `Unknown binding COMMON-LISP:SQRT`

```lisp
(cl:defpackage :coalton-typeclasses
  (:use :coalton :coalton-library))

(cl:in-package :coalton-typeclasses)

(coalton-toplevel

  (define-type Translation (Trans Single-Float Single-Float))

  (declare distance (Translation -> Single-Float))
  (define (distance t)
    (match t
      ((Trans dx dy) (cl:sqrt (+ (* dx dx) (* dy dy)))))))
```

It give the same error when replacing the `defpackage` with `(in-package :coalton-user)`
I'm using `SBCL 2.1.11` with Quicklisp and Coalton updated to today.

**Comment by stylewarning** [comment, marked as answer]
Sorry, sqrt isn't in the standard library. Add this above the `distance` definition:

```
(declare sqrt (Single-Float -> Single-Float))
(define (sqrt x)
  (lisp Single-Float (x)
    (cl:sqrt x)))
```

I'm not exactly sure where sqrt should go in the standard library yet.
